Backends/CLX/medium.lisp

```diff
@@ -92,16 +92,37 @@ region and its clipping pixmap. This is looked up for optimization with region-e
           (warn "Unknown cap style ~S, using :round" clim-shape)
           :round))))
 
+;;; XXX: this should be refactored into a reusable protocol in clim-backend
+;;; with specialization on medium. -- jd 2018-10-31
+(defun line-style-scale (line-style medium)
+  (let ((unit (line-style-unit line-style)))
+    (ecase unit
+      (:normal 1)
+      (:point (/ (graft-width (graft medium))
+                 (graft-width (graft medium) :units :inches)
+                 72))
+      (:coordinate (multiple-value-call #'(lambda (x y) (sqrt (+ (expt x 2) (expt y 2))))
+                     (transform-distance (medium-transformation medium) 0.71 0.71))))))
+
+(defun line-style-effective-thickness (line-style medium)
+  (* (line-style-thickness line-style)
+     (line-style-scale line-style medium)))
+
+(defun line-style-effective-dashes (line-style medium)
+  (let ((scale (line-style-scale line-style medium)))
+    (mapcar #'(lambda (dash) (round (* dash scale)))
+            (line-style-dashes line-style))))
+
 (defmethod (setf medium-line-style) :before (line-style (medium clx-medium))
   (with-slots (gc) medium
     (when gc
       (let ((old-line-style (medium-line-style medium)))
-        (unless (eql (line-style-thickness line-style)
-                     (line-style-thickness old-line-style))
-          ;; this is kind of false, since the :unit should be taken
-          ;; into account -RS 2001-08-24
+        (unless (and (eql (line-style-thickness line-style)
+                          (line-style-thickness old-line-style))
+                     (eql (line-style-unit line-style)
+                          (line-style-unit old-line-style)))
           (setf (xlib:gcontext-line-width gc)
-                (round (line-style-thickness line-style))))
+                (round (line-style-effective-thickness line-style medium))))
```

Comment:
Could `line-style-effective-thickness` also round, for consistency with `line-style-effective-dashes`?

Reply:
it does now. note, that it is clx-specific requirement to round these (renderer nor output-only backends doesn't seem to require that), what is a good indicator, that line-style-effective-* should be part of climb package (and not line-style-scale with these functions working on top of that). Fixed.